### PR TITLE
Sr persistance layer 1

### PR DESCRIPTION
**Pull request by simen-rekkedal**
New Persistance layer for SystemUser

## Description

Wrote yuniql migration scripts to create schema, grants, tables and functions.
Wrote Insert new SystemUser function.

## Related Issue(s)
- #379

## Verification
- [ ] **Your** code builds clean without any errors or warnings
- [ ] Manual testing done (required)
- [ ] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [ ] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: src/Persistance/Configuration/PostgresSQLSettings.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    {
+        /// <summary>
+        /// Gets or Sets the connection string to the postgres db.
+        /// </summary>
+        public string ConnectionString { get; set; }
+
+        /// <summary>
+        /// Gets or Sets the password for the app user for the postgres db.
+        /// </summary>
+        public string AuthorizationDbPwd { get; set; }
```
Comment (acn-dgopa) [inline src/Persistance/Configuration/PostgresSQLSettings.cs:16]: authentication

File: src/Persistance/SystemUserRespository.cs [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+            "@_owned_by_party_id, " +
+            "@_supplier_name, " +
+            "@_supplier_org_no, " +
+            "@_client_id)";
+
+        /// <summary>
+        /// Private helper class to hold the Column names as constant strings to aid in typing SQL commands.
+        /// Prefix with an underscore when using them as input fields to the Functions: see the In_() method.
+        /// </summary>
+        private class Column
```
Comment (acn-dgopa) [inline src/Persistance/SystemUserRespository.cs:34]: Just wondering if this can be a static class instead?

**Review by acn-dgopa — APPROVED** [review]
As we discussed, you can see the possibility on using dependency injection class for mock data setup in development environment

File: src/Core/Models/SystemUser.cs [hunk 3]
```diff
@@ -12,66 +15,79 @@ public class SystemUser
...
+        [JsonPropertyName ("Id")]
         public string Id { get; set; }
 
         /// <summary>
         /// The Title and Description are strings set by the end-user in the Frontend.
+        /// In the db this field is required, but if we use this model as a DTO, we allow null
         /// </summary>
+        [AllowNull]
+        [JsonPropertyName("IntegrationTitle")]
         public string IntegrationTitle { get; set; }
```
[the remark this answers is not in the document]
Comment (simen-rekkedal) [reply src/Core/Models/SystemUser.cs:31]: No, it was removed when Mette redesigned the solution in December, will adjust the comment.

File: src/Core/Models/SystemUserRequestDTO.cs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    {
+        [AllowNull]
+        [JsonPropertyName("PartyId")]
+        /// <summary>
+        /// PartyId is the owning Legal Entity, whether an organization or a privat person
+        /// </summary>
+        public string PartyId { get; set; }
+
+        /// <summary>
+        /// The Title and Description are strings set by the end-user in the Frontend.
```
[the remark this answers is not in the document]
Comment (simen-rekkedal) [reply src/Core/Models/SystemUserRequestDTO.cs:23]: No, it was removed when Mette redesigned the solution in December, will adjust the comment.

File: src/Core/Models/SystemUserUpdateDTO.cs [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        /// The OwnedBy identifies the end-user Organisation, and is fetched from the login Context and
+        /// user party serivces
+        /// Even if this DTO allows null, the db field is of course still required
+        /// </summary>
+        [AllowNull]
+        [JsonPropertyName("OwnedByPartyId")]
+        public string OwnedByPartyId { get; set; }
+
+        /// <summary>
+        /// The Title and Description are strings set by the end-user in the Frontend.
```
[the remark this answers is not in the document]
Comment (simen-rekkedal) [reply src/Core/Models/SystemUserUpdateDTO.cs:35]: No, it was removed when Mette redesigned the solution in December, will adjust the comment.

File: src/Authentication/Services/SystemUserService.cs [hunk 6]
```diff
@@ -29,15 +31,17 @@ public SystemUserService()
         /// to ensure that there is no mismatch if the same partyId creates several new SystemUsers at the same time
         /// </summary>
         /// <returns></returns>
```
[the remark this answers is not in the document]
Comment (simen-rekkedal) [reply src/Authentication/Services/SystemUserService.cs:33]: True, will change

File: src/Authentication/Controllers/SystemUserController.cs [hunk 7]
```diff
@@ -96,12 +96,12 @@ public async Task<ActionResult> SetDeleteFlagOnSystemUser(Guid systemUserId)
...
+        [ProducesResponseType(typeof(SystemUser), StatusCodes.Status200OK)]        
         [ProducesResponseType(StatusCodes.Status404NotFound)]        
-        [HttpPost("{partyId}/{createRequestId}")]
-        public async Task<ActionResult<SystemUser>> CreateSystemUser(SystemUser request)
-        {
-            SystemUser? toBeCreated = await _systemUserService.CreateSystemUser(request);
+        [HttpPost]
+        public async Task<ActionResult<SystemUser>> CreateSystemUser([FromBody] SystemUserRequestDto request)
+        {           
+            SystemUser? toBeCreated = await _systemUserService.CreateSystemUser(request, 1);
```
[the remark this answers is not in the document]
Comment (simen-rekkedal) [reply src/Authentication/Controllers/SystemUserController.cs:104]: Yes, that's true, will change the name

**Pull request merged**